ducc/lib/podman_store.go

```diff
@@ -0,0 +1,502 @@
+	}
+	return nil
+}
+
+func (img Image) IngestConfigFile(CVMFSRepo string) (err error) {
+	Log().WithFields(log.Fields{"action": "Creating config file for the image in podman store"}).Info(img.GetSimpleName())
+	manifest, err := img.GetManifest()
+	if err != nil {
+		LogE(err).Warn("Error in getting the image manifest")
+		return err
+	}
+
+	user := img.User
+	pass, err := GetPassword()
+	if err != nil {
+		LogE(err).Warning("Unable to get the credential for downloading the configuration blog, trying anonymously")
+		user = ""
+		pass = ""
+	}
+
+	configUrl := fmt.Sprintf("%s://%s/v2/%s/blobs/%s",
+		img.Scheme, img.Registry, img.Repository, manifest.Config.Digest)
+
+	token, err := firstRequestForAuth(configUrl, user, pass)
+	if err != nil {
+		LogE(err).Warning("Unable to retrieve the token for downloading config file")
+		return err
+	}
+
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", configUrl, nil)
+	if err != nil {
+		LogE(err).Warning("Unable to create a request for getting config file.")
+		return err
+	}
+	req.Header.Set("Authorization", token)
+	req.Header.Set("Accept", "application/vnd.docker.distribution.manifest.v2+json")
+
+	resp, err := client.Do(req)
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		LogE(err).Warning("Error in reading the body from the configuration")
+		return err
+	}
+
+	//write configuration to a temp file
+	tmpFile, err := ioutil.TempFile("", "configFile")
+	err = ioutil.WriteFile(tmpFile.Name(), []byte(body), 0644)
+	tmpFile.Close()
+	if err != nil {
+		LogE(err).Error("Error in writing to the config file")
+		return err
+	}
+
+	//generate config file path to ingest above temp dir.
+	fname, err := generateConfigFileName(manifest.Config.Digest)
+	if err != nil {
+		LogE(err).Warning("Error in generating config file name")
+		return err
+	}
+
+	imageID := calculateId(manifest.Config.Digest)
+	configFilePath := filepath.Join(rootPath, imageMetadataDir, imageID, fname)
+
+	//Ingest config file
+	err = IngestIntoCVMFS(CVMFSRepo, configFilePath, tmpFile.Name())
+	os.RemoveAll(tmpFile.Name())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (img Image) IngestImageManifest(CVMFSRepo string) (err error) {
+	Log().WithFields(log.Fields{"action": "Creating manifest file for the image in podman store"}).Info(img.GetSimpleName())
+	manifest, err := img.GetManifest()
+	if err != nil {
+		LogE(err).Warn("Error in getting the image manifest")
+		return err
+	}
+	imageID := calculateId(manifest.Config.Digest)
+
+	symlinkPath := filepath.Join(rootPath, imageMetadataDir, imageID, "manifest")
+	targetPath := filepath.Join(".metadata", img.Registry, img.Repository + img.GetReference(), "manifest.json")
+
+	err = CreateSymlinkIntoCVMFS(CVMFSRepo, symlinkPath, targetPath)
+	if err != nil {
+		LogE(err).Error("Error in creating the symlink for manifest.json")
+		return err
+	}
+	return nil
+}
+
+func (img Image) CreateLockFiles(CVMFSRepo, fpath string) (err error) {
```

Comment:
Why do we need lock files?  Shouldn't cvmfs transactions make sure that images are imported in an orderly fashion?

Reply:
Hi @jblomer , these lock files are meant for podman to read. Podman expects lock files in the image store directories. I will add comments for the functions. Their role will be more clearer then